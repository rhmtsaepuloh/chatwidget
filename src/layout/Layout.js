import React from "react";
import { useSelector } from "react-redux";

import "./layout.scss";

const Layout = ({ children }) => {
  const { secondaryColor } = useSelector(
    (state) => state.clientReducers.dataClient
  );
  return (
    <div className='wrapper-chat-widget' style={{ background: secondaryColor }}>
      {children}
    </div>
  );
};

export default Layout;
