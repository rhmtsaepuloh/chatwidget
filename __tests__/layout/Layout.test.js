/* eslint-disable react/react-in-jsx-scope */
import React from "react";
import { shallow } from "enzyme";
import { useSelector } from "react-redux";
import Layout from "../../src/layout/Layout";

const children = <div />;

jest.mock("react-redux", () => ({
  useDispatch: jest.fn(),
  useSelector: jest.fn(),
}));

describe("render correctly layout components", () => {
  test("renders with data selector", () => {
    useSelector.mockImplementation((selector) =>
      selector({
        clientReducers: {
          dataClient: {
            secondaryColor: "#CA1234",
          },
        },
      })
    );
    const wrapper = shallow(<Layout>{children}</Layout>);
    expect(wrapper).toMatchSnapshot();
  });
  test("renders", () => {
    const wrapper = shallow(<Layout>{children}</Layout>);
    expect(wrapper).toMatchSnapshot();
  });
});
