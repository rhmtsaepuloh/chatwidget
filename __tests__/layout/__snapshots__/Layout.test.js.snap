// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly layout components renders 1`] = `ShallowWrapper {}`;

exports[`render correctly layout components renders with data selector 1`] = `ShallowWrapper {}`;
